From 3ac6fb9e8eeea7a67c9c3b37f9fa4e32c1fee513 Mon Sep 17 00:00:00 2001
From: Klaas Freitag <freitag@opensuse.org>
Date: Wed, 26 Dec 2018 22:04:12 +0100
Subject: [PATCH 1/6] Fix build with Qt 5.12: Strange include<QtSql> problem
 that breaks.

Workaround: Use specific includes for Sql classes rather than QtSql.
---
 src/catalogchapter.cpp       | 2 +-
 src/defaultprovider.cpp      | 7 ++++++-
 src/templkataloglistview.cpp | 2 +-
 3 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/src/catalogchapter.cpp b/src/catalogchapter.cpp
index 0724cdd..a35d7ff 100644
--- a/src/catalogchapter.cpp
+++ b/src/catalogchapter.cpp
@@ -17,7 +17,7 @@
 
 #include <QtCore>
 #include <QtGui>
-#include <QtSql>
+#include <QSqlQuery>
 
 #include <QDebug>
 
diff --git a/src/defaultprovider.cpp b/src/defaultprovider.cpp
index 55f6ba7..b253ef9 100644
--- a/src/defaultprovider.cpp
+++ b/src/defaultprovider.cpp
@@ -14,7 +14,10 @@
  *   (at your option) any later version.                                   *
  *                                                                         *
  ***************************************************************************/
-#include <QtSql>
+#include <QSqlQuery>
+#include <QSqlTableModel>
+#include <QSqlRecord>
+
 #include <QFile>
 #include <QTextStream>
 #include <QDebug>
@@ -24,6 +27,8 @@
 #include "doctext.h"
 #include "kraftsettings.h"
 #include "doctype.h"
+#include "kraftdoc.h"
+#include "dbids.h"
 
 #include <klocalizedstring.h>
 
diff --git a/src/templkataloglistview.cpp b/src/templkataloglistview.cpp
index 9c4eaeb..6a3d52a 100644
--- a/src/templkataloglistview.cpp
+++ b/src/templkataloglistview.cpp
@@ -16,7 +16,7 @@
  *                                                                         *
  ***************************************************************************/
 
-#include <QtSql>
+#include <QSqlQuery>
 #include <QDebug>
 #include <QMenu>
 
-- 
2.20.1

